compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/SnapshotInvalidationTracker.skiko.kt

```diff
@@ -76,6 +82,20 @@ internal class SnapshotInvalidationTracker(
         Snapshot.sendApplyNotifications()
         snapshotChanges.perform()
     }
+
+    /**
+     * Runs [block], performing any snapshot changes it generates synchronously.
+     *
+     * See [OwnerSnapshotObserverTest.observeReadsChangedBeforeDisposeEffect] for more details.
+     */
+    inline fun <T> performSnapshotChangesSynchronously(block: () -> T): T {
+        return try {
+            renderingThreadId.value = getCurrentThreadId()
+            block()
```

Comment:
It seems that we can combine `needDraw` and `needLayout` flags and reset it here